lib/phobos/actions/process_message.rb

```diff
@@ -57,10 +57,17 @@ def force_encoding(value)
       def process_message(payload)
         instrument('listener.process_message', @metadata) do
           handler = @listener.handler_class.new
-          preprocessed_payload = handler.before_consume(payload)
+          preprocessed_payload = handler.before_consume(payload, @metadata)
```

Comment:
In order to be backwards compatible here we can do 

```ruby
          preprocessed_payload = begin
            handler.before_consume(payload, @metadata)
          rescue ArgumentError => e
            Phobos.deprecate("before_consume now expects metadata as second argument, please update your consumer. This will not be backwards compatible in the future.")
            handler.before_consume(payload)
          end
```

Reply:
Done!